controller/supportbundle_controller.go

```diff
@@ -0,0 +1,456 @@
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "status.podIP",
+										},
+									},
+								},
+								{
+									Name:  "LONGHORN_SUPPORT_BUNDLE_IMAGE",
+									Value: image,
+								},
+								{
+									Name:  "LONGHORN_SUPPORT_BUNDLE_IMAGE_PULL_POLICY",
+									Value: string(pullPolicy),
+								},
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8080,
+								},
+							},
+						},
+					},
+					ServiceAccountName: sbc.ServiceAccount,
+				},
+			},
+		},
+	}
+
+	_, err := sbc.ds.CreateDeployment(deployment)
+	return err
+}
+
+func (sbc *SupportBundleController) setError(sb *longhorn.SupportBundle, reason string) (*longhorn.SupportBundle, error) {
+	log := getLoggerForSupportBundle(sbc.logger, sb)
+	log.Errorf(reason)
+
+	toUpdate := sb.DeepCopy()
+	sbc.SupportBundleInitialized.False(toUpdate)
+	sbc.SupportBundleInitialized.Message(toUpdate, reason)
+
+	toUpdate.Status.State = types.SupportBundleStateError
+	return sbc.ds.UpdateSupportBundle(toUpdate)
+}
+
+func (sbc *SupportBundleController) setState(sb *longhorn.SupportBundle, state types.SuppportBundleState) (*longhorn.SupportBundle, error) {
+	log := getLoggerForSupportBundle(sbc.logger, sb)
+	log.Debugf("[%s] set state to %s", sb.Name, state)
+
+	toUpdate := sb.DeepCopy()
+	toUpdate.Status.State = state
+	return sbc.ds.UpdateSupportBundle(toUpdate)
+}
+
+func (sbc *SupportBundleController) setReady(sb *longhorn.SupportBundle, filename string, filesize int64) (*longhorn.SupportBundle, error) {
+	logrus.Debugf("[%s] set state to %s", sb.Name, types.SupportBundleStateReadyForDownload)
+	toUpdate := sb.DeepCopy()
+	sbc.SupportBundleInitialized.True(toUpdate)
+	toUpdate.Status.State = types.SupportBundleStateReadyForDownload
+	toUpdate.Status.Progress = 100
+	toUpdate.Status.FileName = filename
+	toUpdate.Status.FileSize = filesize
+	return sbc.ds.UpdateSupportBundle(toUpdate)
+}
+
+func (sbc *SupportBundleController) setProgress(sb *longhorn.SupportBundle, progress int) (*longhorn.SupportBundle, error) {
+	logrus.Debugf("[%s] set progress to %d", sb.Name, progress)
+	toUpdate := sb.DeepCopy()
+	toUpdate.Status.Progress = progress
+	return sbc.ds.UpdateSupportBundle(toUpdate)
+}
+
+func (sbc *SupportBundleController) GetStatus(sb *longhorn.SupportBundle) (*ManagerStatus, error) {
+	podIP, err := manager.GetManagerPodIP(sbc.ds)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("http://%s:8080/status", podIP)
+	httpClient := http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	resp, err := httpClient.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	s := &ManagerStatus{}
+	err = json.NewDecoder(resp.Body).Decode(s)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+type ManagerStatus struct {
+	// phase to collect bundle
```

Comment:
No need to those comments since it's pretty self-explained, and it's also an internal structure.